File: docs/snippets/python/message-with-metadata_main.py
# from microsoft.agents.xml.models import ChatMessage, TextContent
# from microsoft.agents.xml.serialization import MessageSerializer
#
# # Create message with metadata
# message = ChatMessage(
#     role="user",
#     message_id="msg-meta-1",
#     timestamp="2024-01-15T10:30:00Z",
#     contents=[
#         TextContent(text="Hello!")
#     ]
# )
#
# serializer = MessageSerializer()
# xml_output = serializer.serialize(message)
# print(xml_output)

File: snippets/csharp/message-with-metadata_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;

// Create message with metadata
var message = new UserMessage
{
    MessageId = "msg-meta-1",
    CreatedAt = DateTime.Parse("2024-01-15T10:30:00Z").ToUniversalTime(),
    Contents = new List<AIContent>
    {
        new TextContent { Text = "Hello!" }
    }
};

var serializer = new MessageSerializer();
var xmlOutput = serializer.Serialize(message);
Console.WriteLine(xmlOutput);